### PR TITLE
Modified the model for multiple hidden layers

### PR DESCRIPTION
**Pull request by ShubhamShaswat**
I have added a new class to implement multi layer percepton. 

- The user could able to build model by passing a list of hidden number of units (ex: `num_hidden=[128,128]`) for each hidden layers.

- I have also added a function `summary()` to get details of each layer of the model.

### CODE REVIEW

**Comment by ShubhamShaswat** [comment]
@skoudoro Thanks for the suggestion,I have restored the old model along with the new one

**Comment by skoudoro** [comment]
Hi @ShubhamShaswat,

- Can you fix the last pep8 as you can see in this link: https://app.codacy.com/manual/dipy/dipy/pullRequest?prid=5216162

- Can you add a docstring to all methods, please? (`fit`, `evaluate`, `summary`, `predict`)

@ShreyasFadnavis, @Garyfallidis, @arokem, or any neural network expert, can you look at this PR. 

Thank you

**Comment by ShubhamShaswat** [comment]
Hi @skoudoro, thanks again for the suggestion. Adding docstring will definitely be helpful.

**Comment by skoudoro** [comment]
@ShreyasFadnavis, @Garyfallidis, @arokem, or any neural network expert, can you look at this PR.
Any opinions?

Thank you

File: dipy/nn/model.py [hunk 1]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
         self.model = model
 
     def fit(self, x_train, y_train, epochs=5):
+
+        """train the model on train dataset.
```
Comment (Garyfallidis) [inline dipy/nn/model.py:59]: Train

File: dipy/nn/model.py [hunk 2]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+        """
+
         hist = self.model.fit(x_train, y_train, epochs=epochs)
         self.accuracy = hist.history['accuracy'][0]
         self.loss = hist.history['loss'][0]
         return hist
 
     def evaluate(self, x_test, y_test, verbose=2):
+
+        """evaluate the model on test dataset.
```
Comment (Garyfallidis) [inline dipy/nn/model.py:89]: Evaluate ( just start with a capital letter on the docstring)

File: dipy/nn/model.py [hunk 3]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+        -------
+        evaluate : List
+            return list of loss value and accuracy value on test dataset
+
+        """
         return self.model.evaluate(x_test,  y_test, verbose=verbose)
 
     def predict(self, x_test):
+
+        """predict the output from input samples.
```
Comment (Garyfallidis) [inline dipy/nn/model.py:113]: as above

File: dipy/nn/model.py [hunk 4]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+
+    def __init__(self, input_shape=(28,28),
+                        num_hidden=[128],
+                        act_hidden='relu',
+                        dropout=0.2,
+                        num_out=10,
+                        act_out='softmax',
+                        loss='sparse_categorical_crossentropy',
+                        optimizer='adam'):
+
```
Comment (Garyfallidis) [inline dipy/nn/model.py:143]: too many spaces

File: dipy/nn/model.py [hunk 5]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+                Number of nodes in output layer
+            act_out : string
+                Activation function used in output layer
+            optimizer :  string
+                Select optimizer. Default adam.
+            loss : string
+                Select loss function for measuring accuracy.
+                Default sparse_categorical_crossentropy.
+        """
+
```
Comment (Garyfallidis) [inline dipy/nn/model.py:167]: remove empty line

File: dipy/nn/model.py [hunk 6]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+        self.input_shape = input_shape
+        self.num_hidden = num_hidden
+        self.act_hidden = act_hidden
+        self.dropout = dropout
+        self.num_out = num_out
+        self.act_out = act_out
+        self.loss = loss
+        self.optimizer = optimizer
+        self.accuracy = None
+
```
Comment (Garyfallidis) [inline dipy/nn/model.py:178]: remove empty lines (see pep8)

File: dipy/nn/model.py [hunk 7]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+        for i in range(len(self.num_hidden)):
+            x = tf.keras.layers.Dense(self.num_hidden[i])(x)
+
+        x = tf.keras.layers.Dropout(self.dropout)(x)
+        out = tf.keras.layers.Dense(self.num_out, activation=self.act_out)(x)
+
+        self.model = tf.keras.models.Model(inputs=inp, outputs=out)
+
+
```
Comment (Garyfallidis) [inline dipy/nn/model.py:193]: same

File: dipy/nn/model.py [hunk 8]
```diff
@@ -56,13 +55,238 @@ def __init__(self, input_shape=(28, 28),
...
+        summary :  NoneType
+            the summary of the model
+
+        """
+        return self.model.summary()
+
+
+    def fit(self, x_train, y_train, epochs=5):
+
+        """train the model on train dataset.
```
Comment (Garyfallidis) [inline dipy/nn/model.py:220]: Start with capital letter

**Comment by ShubhamShaswat** [comment]
Thanks @Garyfallidis for the review,I have modify the file with the requested changes.

File: dipy/nn/model.py [hunk 9]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
 
         self.model = model
 
+    def summary(self):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:58]: Can you remove empty line?

File: dipy/nn/model.py [hunk 10]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        Returns
+        -------
+        summary :  NoneType
+            the summary of the model
+
+        """
+        return self.model.summary()
+
     def fit(self, x_train, y_train, epochs=5):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:74]: Can you remove empty line?

File: dipy/nn/model.py [hunk 11]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+            the number of epochs
+
+        Returns
+        -------
+        hist : object
+            A History object. Its History.history attribute is a record of
+            training loss values and metrics values at successive epochs
+
+        """
+
```
Comment (skoudoro) [inline dipy/nn/model.py:96]: Can you remove empty line?

File: dipy/nn/model.py [hunk 12]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
         hist = self.model.fit(x_train, y_train, epochs=epochs)
         self.accuracy = hist.history['accuracy'][0]
         self.loss = hist.history['loss'][0]
         return hist
 
     def evaluate(self, x_test, y_test, verbose=2):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:103]: Can you remove empty line?

File: dipy/nn/model.py [hunk 13]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        Returns
+        -------
+        evaluate : List
+            return list of loss value and accuracy value on test dataset
+
+        """
         return self.model.evaluate(x_test,  y_test, verbose=verbose)
 
     def predict(self, x_test):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:127]: Can you remove empty line?

File: dipy/nn/model.py [hunk 14]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+    def __init__(self, input_shape=(28,28),
+                 num_hidden=[128],
+                 act_hidden='relu',
+                 dropout=0.2,
+                 num_out=10,
+                 act_out='softmax',
+                 loss='sparse_categorical_crossentropy',
+                 optimizer='adam'):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:156]: Can you remove empty line?

File: dipy/nn/model.py [hunk 15]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        self.model = tf.keras.models.Model(inputs=inp, outputs=out)
+
+        #compiling the model
+        self.model.compile(optimizer=self.optimizer,
+                              loss=self.loss,
+                              metrics=['accuracy'])
+
+    def summary(self):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:209]: Can you remove empty line?

File: dipy/nn/model.py [hunk 16]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        Returns
+        -------
+        summary :  NoneType
+            the summary of the model
+
+        """
+        return self.model.summary()
+
+    def fit(self, x_train, y_train, epochs=5):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:225]: Can you remove empty line?

File: dipy/nn/model.py [hunk 17]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        Returns
+        -------
+        hist : object
+            A History object. Its History.history attribute is a record of
+            training loss values and metrics values at successive epochs
+
+
+        """
+
```
Comment (skoudoro) [inline dipy/nn/model.py:248]: Can you remove empty line?

File: dipy/nn/model.py [hunk 18]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        epochs : int (Default = 5)
+            the number of epochs
+
+        Returns
+        -------
+        hist : object
+            A History object. Its History.history attribute is a record of
+            training loss values and metrics values at successive epochs
+
+
```
Comment (skoudoro) [inline dipy/nn/model.py:246]: too many empty lines, keep only one

File: dipy/nn/model.py [hunk 19]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        -------
+        evaluate : List
+            return list of loss value and accuracy value on test dataset
+
+        """
+
+        return self.model.evaluate(x_test,  y_test, verbose=verbose)
+
+    def predict(self, x_test):
+
```
Comment (skoudoro) [inline dipy/nn/model.py:280]: Can you remove empty line?

File: dipy/nn/tests/test_tf.py [hunk 20]
```diff
@@ -61,6 +61,29 @@ def test_default_mnist_slp():
     assert_equal(x_test_prob.shape, (10000, 10))
 
 
+@pytest.mark.skipif(not have_tf, reason='Requires TensorFlow')
+def test_default_mnist_mlp():
+
```
Comment (skoudoro) [inline dipy/nn/tests/test_tf.py:66]: Can you remove empty line?

File: dipy/nn/model.py [hunk 21]
```diff
@@ -55,14 +54,244 @@ def __init__(self, input_shape=(28, 28),
...
+        dataset.
+
+        Parameters
+        ----------
+        x_test : ndarray
+            the x_test is the test dataset
+        y_test : ndarray shape=(BatchSize,)
+            the y_test is the labels of the test dataset
+        verbose : int (Default = 2)
+            By setting verbose 0, 1 or 2 you just say how do you want to 'see' the training progress for each epoch.
```
Comment (skoudoro) [inline dipy/nn/model.py:268]: pep8 : too long line

**Comment by ShubhamShaswat** [comment]
Thanks @skoudoro for reviewing my code.Glad to help

**Comment by skoudoro** [comment]
Thank you for this @ShubhamShaswat. Merging

**Pull request merged**